### PR TITLE
Require using UnmanagedCallConv instead of the CallingConvention field on GeneratedDllImport

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Block specifying CallingConvention on GeneratedDllImport and update the converter to automatically create the UnmanagedCallConv attribute.

### CODE REVIEW

File: DllImportGenerator/DllImportGenerator/DllImportGenerator.cs [hunk 1]
```diff
@@ -124,6 +124,11 @@ public void Execute(GeneratorExecutionContext context)
                     generatorDiagnostics.ReportConfigurationNotSupported(generatedDllImportAttr, nameof(DllImportStub.GeneratedDllImportData.ThrowOnUnmappableChar));
                 }
 
+                if (stubDllImportData!.IsUserDefined.HasFlag(DllImportStub.DllImportMember.CallingConvention))
```
Comment (AaronRobinsonMSFT) [inline DllImportGenerator/DllImportGenerator/DllImportGenerator.cs:127]: We should update out design documents - specifically compat with this.

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: designs/Compatibility.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ The built-in system treats `CharSet.None` as `CharSet.Ansi`. The P/Invoke source
...
 
+[`CallingConvention`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.dllimportattribute.callingconvention) will not be supported for `GeneratedDllImportAttribute`. Users will be required to use the new `UnmanagedCallConvAttribute` attribute instead. This attribute provides support for extensible calling conventions and provides parity with the `UnmanagedCallersOnlyAttribute` attribute and C# function pointer syntax. We will enable our conversion code-fix to automatically convert explicit calling convention usage to use the `UnmanagedCallConvAttribute`.
```
Comment (AaronRobinsonMSFT) [inline designs/Compatibility.md]:
```suggestion
[`CallingConvention`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.dllimportattribute.callingconvention) will not be supported for `GeneratedDllImportAttribute`. Users will be required to use the new `UnmanagedCallConvAttribute` attribute instead. This attribute provides support for extensible calling conventions and provides parity with the `UnmanagedCallersOnlyAttribute` attribute and C# function pointer syntax. We will enable our conversion code-fix to automatically convert explicit and known calling convention usage to use the `UnmanagedCallConvAttribute`.
```

File: DllImportGenerator/DllImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs [hunk 3]
```diff
@@ -200,11 +211,74 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            unmanagedCallConvAttribute = null;
+            if (editor.SemanticModel.Compilation.GetTypeByMetadataName(TypeNames.UnmanagedCallConvAttribute) is null)
+            {
+                return false;
+            }
+
+            SyntaxNode attribute = generator.Attribute(TypeNames.UnmanagedCallConvAttribute);
+            if (callingConvention == CallingConvention.Winapi)
+            {
+                unmanagedCallConvAttribute = attribute;
```
Comment (elinor-fung) [inline DllImportGenerator/DllImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs]: Can we just drop setting the `CallingConvention` without needing to add `UnmanagedCallConv` in this case?

**Review by elinor-fung — APPROVED** [review]

File: DllImportGenerator/DllImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs (hunk 3 above)
Comment (jkoritzinsky) [reply DllImportGenerator/DllImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs]: Yes we can. I'll update the change to do that.

**Pull request merged**